### ISSUE TITLE
realm WriteCopyTo backup works - opening file failes with decryption failed

### ISSUE THREAD

**Issue by pmahend1** [realm/realm-dotnet#1779]
## Goals

When I try to open encrypted backed up realm file from Realm studio, with the same encryption key that it was configured  it says decryption failed.

```Unable to open a realm at path 'C:\Users\Foo\Desktop\123.realm': Realm file decryption failed.```

## Expected Results

Should be able to open realm file.

## Actual Results

decryption failes even though file was not suspended realm file.

```Unable to open a realm at path 'C:\Users\Foo\Desktop\123.realm': Realm file decryption failed.```

## Code Sample

Using  
```

            var realmInstance = Realm.GetInstance(config);
            var configNew = new RealmConfiguration(Path.GetTempFileName())
            {
                EncryptionKey = realmInstance.Config.EncryptionKey
            };
            File.Delete(configNew.DatabasePath);

            realmInstance.Write(() =>
            {
                realmInstance.WriteCopy(configNew);
            });
            realmInstance.WriteCopy(configNew);
```

**Key**
```string planKey= BitConverter.ToString(realmInstance.Config.EncryptionKey).Replace("-", string.Empty);```

Another method tried to backup realm file.
```
realmInstance.Dispose();
```
And 

Copying the file at ```realmInstance.Config.DatabasePath``` with Java native stream method. 

 Still the same error

## Version of Realm and Tooling
using Xamarin Forms **Realm Nuget package v 3.2.1**
Realm Studio - 2.9.0

- Realm Object Server Version: ?
- Flavor:
  - [x] Developer
  - [ ] Professional
  - [ ] Enterprise
- Server OS & Version: ? Using Realm database
- Client SDK Version: ? NA
- Client OS & Version: ?Windows for Realm Studio , Android for .realm

**Comment by fealebenpae** [comment]
Are you trying to open an encrypted realm file created on an Android device in Studio running on Windows?

**Comment by pmahend1** [comment]
Yes. I mailed it to myself from Android share feature. Want to check if the backup is valid

**Comment by fealebenpae** [comment]
Unfortunately encrypted realms are not cross-platform - you will not be able to open the file on Windows.

**Comment by pmahend1** [comment]
Ok. Can I still open it within Android in backend?

So you are saying even though it's Realm file I can't open the same file in iOS if I had cross platform xamarin apps using same realm database.?

**Comment by fealebenpae** [comment]
Yes, you can open the file on the same Android device.

The way Realm encryption works is dependent on the size of a memory page - usually that's different between 32bit and 64bit devices (or more specifically the architecture of the app, not the device).

I supposed you can open the same encrypted realm file created on a 64bit Android device on a 64bit iOS device, provided your app is running in the same architecture in both cases, but if you want to share encrypted realm files between devices you should really take the hardware difference into account.

Unencrypted realm files can be open on any platform, regardless of where they were first created.

**Comment by pmahend1** [comment]
FYI I tried to open in it from these two Android apps,

- [Browser For Realm Mobile Database](https://play.google.com/store/apps/details?id=com.min60.android.realmbrowser.free)

> Wrong encryption key

- [Realm Browswer](https://play.google.com/store/apps/details?id=com.min60.android.realmbrowser.free)

> Reasons for Realm file open failure
> 1. Not a Realm File 
> 2. Insufficient permissions 
> 3. File corrupted
> 4. Encrypted with password 

1. Not a Realm File - it is realm file. Copied from  WriteCopy
2. Insufficient permissions - File is stored in Downloads directory not in the internal directory where root access is required.
3. File corrupted - Is it?
4. Encrypted with password - But the app provides adding key for unlocking.

**Comment by fealebenpae** [comment]
Neither of those apps are developed or maintained by Realm. I suggest creating a blank Android app with the Realm SDK, loading the realm file in it, and checking if the app can open the realm with the encryption key.

**Comment by pmahend1** [comment]
> Neither of those apps are developed or maintained by Realm.  

Thought so.

>  suggest creating a blank Android app with the Realm SDK, loading the realm file in it, and checking if the app can open the realm with the encryption key.

Its working , it can create an instance with this configuration  with copied Realm file location, and able to query values. 

- [ ] My intention is to provide ability to back up .realm db file and restore it within app. I will check downloading realm file back to android and chosing with file picker and read it

**Comment by pmahend1** [comment]
I was able to fix this by using ```Realm.WriteCopyTo()``` while sharing via intent within Android.  I guess this keeps integrity rather than using Java native input output byte stream to copy files. 

Then loading new realm file with copied configuration. For some reason had ```{System.InvalidOperationException: No RealmObjects. Has linker stripped them? See https://realm.io/docs/xamarin/latest/#linker-stripped-schema``` , I moved the Realm open code to project which has realm nuget and has fody file. Not sure if that fixed this linker issue.

I made sure the app had storage write read/write access too.
